plugins/org.eclipse.n4js/src/org/eclipse/n4js/typesystem/utils/TypeAliasComputer.xtend

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2021 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.typesystem.utils
+
+import com.google.inject.Inject
+import org.eclipse.n4js.ts.typeRefs.ParameterizedTypeRef
+import org.eclipse.n4js.ts.typeRefs.TypeArgument
+import org.eclipse.n4js.ts.typeRefs.TypeRef
+import org.eclipse.n4js.ts.typeRefs.TypeRefsFactory
+import org.eclipse.n4js.ts.types.TypeAlias
+import org.eclipse.n4js.ts.utils.TypeUtils
+import org.eclipse.n4js.typesystem.N4JSTypeSystem
+import org.eclipse.n4js.utils.RecursionGuard
+
+/**
+ * Type system helper strategy for dealing with {@link TypeAlias type aliases}.
+ */
+package class TypeAliasComputer extends TypeSystemHelperStrategy {
+
+	@Inject
+	private N4JSTypeSystem ts;
+
+	// FIXME API doc
+	def package TypeRef resolveTypeAliasFlat(RuleEnvironment G, TypeRef typeRef) {
+		if (!typeRef.isAliasUnresolved) {
+			return typeRef;
+		}
+
+		var ParameterizedTypeRef originalAliasTypeRef = null;
+		var RecursionGuard<TypeAlias> guard = null;
+		var currTypeRef = typeRef;
+		while (true) {
```

Comment:
True. I also had the feeling there should be a more elegant way, but couldn't think of anything better. But I think readability is ok-ish and the impact of this is local to the body of this one method, so no major harm done, I guess.

In any case, my main concern was to avoid recursive invocations, here, to not bloat the stack even more than is done by the type system already.